pkg/source/reconciler/mtsource/kafkasource.go

```diff
@@ -183,3 +188,46 @@ func (r *Reconciler) createCloudEventAttributes(src *v1beta1.KafkaSource) []duck
 	}
 	return ceAttributes
 }
+
+// initPolicyFromConfigMap reads predicates and priorities data from configMap
+func initPolicyFromConfigMap(ctx context.Context, configMapName string, policy *sts.SchedulerPolicy) error {
+	policyConfigMap, err := kubeclient.Get(ctx).CoreV1().ConfigMaps(system.Namespace()).Get(context.TODO(), configMapName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("couldn't get scheduler policy config map %s/%s: %v", system.Namespace(), configMapName, err)
+	}
+
+	preds, found := policyConfigMap.Data["predicates"]
+	if !found {
+		return fmt.Errorf("missing policy config map value at key predicates")
+	}
+	if err := json.NewDecoder(strings.NewReader(preds)).Decode(&policy.Predicates); err != nil {
+		return fmt.Errorf("invalid policy: %v", err)
+	}
+	logging.FromContext(ctx).Infof("Predicates to be registered: %v", policy.Predicates)
+
+	priors, found := policyConfigMap.Data["priorities"]
+	if !found {
+		return fmt.Errorf("missing policy config map value at key priorities")
+	}
+	if err := json.NewDecoder(strings.NewReader(priors)).Decode(&policy.Priorities); err != nil {
+		return fmt.Errorf("invalid policy: %v", err)
+	}
+	logging.FromContext(ctx).Infof("Priorities to be registered: %v", policy.Priorities)
+
+	if errs := validatePolicy(policy); errs != nil {
+		return errors.New("validating policy failed")
```

Comment:
you may want to return `errs` or at least log them

Reply:
Done.